aten/src/ATen/native/BatchLinearAlgebra.cpp

```diff
@@ -256,6 +256,38 @@ TORCH_META_FUNC(linalg_lu_factor_ex)(const Tensor& A, bool pivot, bool check_err
   set_output(2, sizes, {}, A.options().dtype(kInt), {});
 }
 
+TORCH_META_FUNC(linalg_svd)(const Tensor& A,
+                            bool full_matrices) {
+  TORCH_CHECK(A.dim() >= 2, "torch.linalg.svd: input should have at least 2 dimensions, but has ", A.dim(), " dimensions instead");
+
+  // We need to distinguish the cuSOLVER case, as cuSOLVER expects F-contig matrices, but
+  // it computes V rather than Vh
+  const bool use_cusolver = at::native::svd_uses_cusolver(A);
+
+  auto sizes = A.sizes().vec();
+  const auto m = sizes.cend()[-2];
+  const auto n = sizes.cend()[-1];
+  const auto k = std::min(m, n);
+
+  // Prepare sizes for U
+  sizes.back() = full_matrices ? m : k;
+  // Optimization: If A is C-contig (F-transposed) we compute the SVD of A^T and transpose the outputs
```

Comment:
> we compute the SVD of A^T

Could you please point to the code where it's done? I think you use column-major copy both on CPU and CUDA.